guides/common/modules/con_puppet-architecture.adoc

```diff
@@ -1,15 +1,30 @@
-[id="puppet_guide_puppet_architecture_{context}"]
+[id="Puppet_architecture_{context}"]
 = Puppet Architecture
 
-Puppet uses a server-client architecture with managed hosts running the Puppet agent and {Project} or any {SmartProxies} acting as a Puppet server.
-The Puppet agent enforces the declared state on managed hosts.
-It sends facts to the Puppet server, receives catalogs that are compiled from Puppet manifests, and sends reports to your {Project}.
-Differences between the declared state and the actual state are called _drifts_ in Puppet terminology.
+Puppet is built on client-server architecture.
+The Puppet server is the central component, where the configuration definitions live.
+{ProjectServer} or any {SmartProxies} are typically deployed with the Puppet server and {Project} acts as an https://puppet.com/docs/puppet/7/nodes_external.html[External Node Classifier (ENC)] for such Puppet server.
+Managed hosts run the Puppet agent as a client that communicates with the Puppet server.
 
-Puppet facts are information about the hosts running the Puppet agent.
-Running `puppet facts` on a host displays Puppet facts in JSON format.
-They are collected by the Puppet agent and reported to the Puppet server on each run.
+The Puppet agent collects _facts_ about a host and reports them to the Puppet server on each run.
+You can display the Puppet _facts_ in JSON format by running `puppet facts` on a host.
 
-{Project} users can define Puppet parameters as _key-value_ pairs which behave similar to host parameters or Ansible variables.
+The Puppet server forwards _facts_ to {Project} and {Project} stores them for later use.
+Based on the _facts_ and other definitions, {Project} constructs the ENC answer to the Puppet server.
+The Puppet server compiles a catalog based on the ENC answer and sends the catalog to the Puppet agent.
+The Puppet agent evaluates the system state on the host.
+If the Puppet agent finds differences (_drifts_) between the defined state in the catalog and the actual state,
+it enforces correction of the host's state.
+The Puppet agent then reports correction results back to the Puppet server, which reports them to {Project}.
 
-This section is set out to use a Puppet module to install, configure, and manage the _ntp_ service.
+The system state of a host is defined by means of the catalog.
+A catalog is compiled from Puppet manifests of one or more Puppet modules assigned to the host.
+A Puppet module is a collection of classes, manifests, resources, files, and templates.
```

Comment:
```suggestion
Puppet module:: 
A collection of classes, manifests, resources, files, and templates.
```
A definition list for all the new terms would probably be easier on the eye and easier for scanning.

Reply:
@melcorr Would this work?